### PR TITLE
Backend/set up database

### LINKED ISSUE #65: Spike database in the cloud

### PR DESCRIPTION
**Pull request by jmir17**
Resolves #65 #67 #69 #70 #72
to run the container generated:
`docker run -it -e SPRING_ENV=prod -e PROD_DB_PASSWD=ask-the-password codurance/open-space-backend:ecb589c451dad56bba93fc476464d739122db595`

### CODE REVIEW

**Comment by sherlockq** [comment]
1. Since the CircleCI and terraform script don't adapt to the change, this pr will break our AWS environment
2. the postgres link of dev profile in application.yml won't work if you start backend outside docker-compose; I suggest that one move profile created like this:
```
---

spring:
  profiles: dev
  datasource:
    driver-class-name: org.postgresql.Driver
    url: "jdbc:postgresql://localhost:5432/open-space"
    username: postgres
    password: example
    initialization-mode: always

---

spring:
  profiles: dev-in-docker
  datasource:
    driver-class-name: org.postgresql.Driver
    url: "jdbc:postgresql://postgres:5432/open-space"
    username: postgres
    password: example
    initialization-mode: always

---
```

3. In an offline discussion, we concluded that the username/password for db may be put on circleci, and inject into terraform script like this 
```
# in ecs-task-definition.json.tpl
    "environment" : [
      { "name" : "SPRING_ENV", "value" : "prod" },
      { "name" : "PROD_DB_PASSWD", "value" : "asdfasdf" }
    ]
```

File: packages/backend/src/main/resources/application.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ spring:
   profiles: dev
   datasource:
     driver-class-name: org.postgresql.Driver
-    url: "jdbc:postgresql://localhost:5432/open-space"
+    url: "jdbc:postgresql://postgres:5432/open-space"
     username: postgres
```
Comment (haletothewood) [inline packages/backend/src/main/resources/application.yml]: This is weirdly the exact same commit as the previous commit 👀

**Comment by haletothewood** [comment]
> 1. Since the CircleCI and terraform script don't adapt to the change, this pr will break our AWS environment
> 2. the postgres link of dev profile in application.yml won't work if you start backend outside docker-compose; I suggest that one move profile created like this:
> 
> ```
> ---
> 
> spring:
>   profiles: dev
>   datasource:
>     driver-class-name: org.postgresql.Driver
>     url: "jdbc:postgresql://localhost:5432/open-space"
>     username: postgres
>     password: example
>     initialization-mode: always
> 
> ---
> 
> spring:
>   profiles: dev-in-docker
>   datasource:
>     driver-class-name: org.postgresql.Driver
>     url: "jdbc:postgresql://postgres:5432/open-space"
>     username: postgres
>     password: example
>     initialization-mode: always
> 
> ---
> ```
> 
> 1. In an offline discussion, we concluded that the username/password for db may be put on circleci, and inject into terraform script like this
> 
> ```
> # in ecs-task-definition.json.tpl
>     "environment" : [
>       { "name" : "SPRING_ENV", "value" : "prod" },
>       { "name" : "PROD_DB_PASSWD", "value" : "asdfasdf" }
>     ]
> ```

Would love an explanation at some point of why this would break prod env. Also, I don't see anything in Terraform I anticipated something along the lines of https://www.terraform.io/docs/providers/aws/r/db_instance.html

**Comment by haletothewood** [comment]
Cracking work by the way 🎉 .

**Comment by haletothewood** [comment]
This works. It's been fully tested by redeploying backend service after creating a post and the post is persisted. Terraform works also so CI/CD should be all gravy.

**Pull request merged**